In [1]:
import pandas as pd
import numpy as np

In [2]:
data=pd.read_csv("Merged_Landslide_Data.csv")

In [15]:
data.shape

(48108, 52)

In [3]:
data.dtypes

Event_ID                          str
Date                              str
State                             str
District                          str
Latitude                      float64
Longitude                     float64
Season                            str
Rainfall_mm                   float64
Elevation_m                   float64
Slope_Degree                  float64
Soil_Type                         str
NDVI                          float64
Temperature_C                 float64
Humidity                      float64
Distance_to_River_km          float64
Land_Use                          str
Historical_Landslide_Count    float64
Landslide_Occurred                str
Landslide_Risk                    str
Casualties                    float64
Injured                       float64
Missing_Persons               float64
Families_Affected             float64
Population_Affected           float64
Houses_Damaged                float64
Roads_Blocked_km              float64
Bridges_Dama

In [4]:
data["Date"] = pd.to_datetime(data["Date"], errors="coerce")

In [18]:
data["Date"].isna().sum()

np.int64(892)

In [5]:
data.dtypes

Event_ID                                 str
Date                          datetime64[us]
State                                    str
District                                 str
Latitude                             float64
Longitude                            float64
Season                                   str
Rainfall_mm                          float64
Elevation_m                          float64
Slope_Degree                         float64
Soil_Type                                str
NDVI                                 float64
Temperature_C                        float64
Humidity                             float64
Distance_to_River_km                 float64
Land_Use                                 str
Historical_Landslide_Count           float64
Landslide_Occurred                       str
Landslide_Risk                           str
Casualties                           float64
Injured                              float64
Missing_Persons                      float64
Families_A

In [11]:
print(data["Date"].isna().sum())

892


In [13]:
invalid_dates = data[data["Date"].isna()]
invalid_dates

,Event_ID,Date,State,District,Latitude,Longitude,Season,Rainfall_mm,Elevation_m,Slope_Degree,...,Cranes,Relief_Camps,Evacuated_People,Aid_Amount_INR,Relief_Materials_Tons,Compensation_Provided,Recovery_Days,Power_Outage_Hours,Water_Supply_Disrupted,Communication_Disruption
6,LS-009124,NaT,West Bengal,Darjeeling,27.02549,88.59644,Pre-Monsoon,117.2,2538.5,32.2,...,0.0,0.0,14.0,0.0,0.0,No,2.0,0.0,No,No
14,LS-030514,NaT,Nagaland,Phek,26.75895,93.98653,Winter,44.0,1288.7,35.4,...,0.0,0.0,6.0,0.0,0.0,No,2.0,0.0,No,No
17,LS-020002,NaT,Nagaland,Kohima,26.34676,93.70119,Monsoon,351.1,953.0,42.2,...,0.0,0.0,7.0,0.0,0.0,No,2.0,0.0,No,No
56,LS-026122,NaT,Kerala,Kottayam,10.33130,76.40348,Winter,19.4,20.0,50.0,...,0.0,0.0,11.0,0.0,0.0,No,1.0,0.0,No,No
88,LS-039022,NaT,Kerala,Kottayam,11.96685,77.05378,Pre-Monsoon,78.5,1032.2,32.9,...,0.0,0.0,12.0,0.0,0.0,No,0.0,0.0,No,No
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
47668,LS-007573,NaT,Sikkim,Gangtok,27.71661,88.08830,Post-Monsoon,161.8,119.7,30.7,...,1.0,1.0,31.0,878880.0,14.1,Yes,239.0,0.2,Yes,No
47812,LS-043111,NaT,Himachal Pradesh,Kullu,32.46528,77.90948,Pre-Monsoon,79.1,1524.8,40.5,...,0.0,0.0,18.0,0.0,0.0,No,2.0,0.0,No,No
47932,LS-015403,NaT,Uttarakhand,Pithoragarh,29.90715,79.36813,Post-Monsoon,142.2,528.1,6.4,...,0.0,0.0,1.0,0.0,0.0,No,2.0,0.0,No,No
48069,LS-043409,NaT,Goa,South Goa,15.50546,74.00469,Winter,NaN,1451.1,20.0,...,4.0,2.0,71.0,NaN,11.1,No,106.0,31.2,Yes,No


In [22]:
response = pd.to_numeric(data["Response_Time_Min"], errors="coerce")

print((response.isna() & data["Response_Time_Min"].notna()).sum())

0


In [24]:
print(data["Response_Time_Min"].isna().sum())

1555


In [26]:
print(data["Historical_Landslide_Count"].min())
print(data["Historical_Landslide_Count"].max())
print((data["Historical_Landslide_Count"] < 0).sum())

0.0
12.0
0


In [28]:
print(data["Houses_Damaged"].min())
print(data["Houses_Damaged"].max())
print((data["Houses_Damaged"] < 0).sum())

0.0
65.0
0


In [29]:
print(data["Bridges_Damaged"].min())
print(data["Bridges_Damaged"].max())
print((data["Bridges_Damaged"] < 0).sum())

0.0
5.0
0


In [33]:
print(data["State"].str.strip().str.title().unique())
print("======================================================")
print(data["State"].str.strip().str.title().nunique())

<StringArray>
[          'Manipur',  'Himachal Pradesh', 'Jammu And Kashmir',
          'Nagaland',       'Uttarakhand',       'West Bengal',
            'Sikkim',            'Kerala', 'Arunachal Pradesh',
               'Goa',             'Assam',        'Tamil Nadu',
            'Odisha',         'Meghalaya',       'Maharashtra',
    'Andhra Pradesh',           'Mizoram',                 nan,
         'Karnataka']
Length: 19, dtype: str
18


In [36]:
print(data["Soil_Type"].str.strip().str.title().unique())
print("======================================================")
print(data["Soil_Type"].str.strip().str.title().nunique())

<StringArray>
['Loamy', 'Laterite', 'Alluvial', 'Rocky', 'Sandy Loam', 'Silty', 'Clayey',
 nan]
Length: 8, dtype: str
7


In [42]:
print(data["Human_Resources_Deployed"].isna().sum())

2217


In [38]:
print(data["Human_Resources_Deployed"].nunique())

95


In [41]:
print(data["Human_Resources_Deployed"].dtype)

float64


In [43]:
print(data["Human_Resources_Deployed"].describe())

count    45891.000000
mean        17.973023
std         17.519020
min          2.000000
25%          6.000000
50%         11.000000
75%         27.000000
max        101.000000
Name: Human_Resources_Deployed, dtype: float64


## Exact findings from the raw CSV

- Rows profiled: **48,108**
- Columns profiled: **52**
- Date: **0 malformed values**, **892** blanks kept as **NULL**
- **Response_Time_Min**: **0 malformed numeric tokens**, **1555** blanks kept as **NULL**
- **Historical_Landslide_Count**: min **0**, max **12**, invalid rows **0**
- **Houses_Damaged**: min **0**, max **65**, invalid rows **0**
- **Bridges_Damaged**: min **0**, max **5**, invalid rows **0**
- **State**: **18** labels, no inconsistency after trim/case normalization
- **Soil_Type**: **7** labels, no inconsistency after trim/case normalization
- **Human_Resources_Deployed**: It contains **2217** missing values (NaN) and 95 unique numeric levels. No malformed text values were detected because the column is already numeric.
